skt/executable.py

```diff
@@ -1026,61 +311,25 @@ def load_config(args):
     return cfg
 
 
-def check_args(parser, args):
-    # pylint: disable=protected-access
-    """Check the arguments provided to ensure all requirements are met.
-
-    Args:
-      parser - the parser object
-      args   - the parsed arguments
-    """
-    # Users must specify a glob to match generated kernel config files when
-    # building configs with `make rh-configs`
-    if (args._name == 'build' and args.cfgtype == 'rh-configs'
-            and not args.rh_configs_glob):
-        parser.error("--cfgtype rh-configs requires --rh-configs-glob to set")
-
-    # Check required arguments for 'report'
-    if args._name == 'report':
-
-        # MailReporter requires recipient and sender email addresses
-        if (args.type == 'mail' and (not args.mail_to or not args.mail_from)):
-            parser.error(
-                "--reporter mail requires --mail-to and --mail-from to be set"
-            )
-
-        # No mail-related options should be provided for StdioReporter
-        if (args.type == 'stdio'
-                and any([getattr(args, x)
-                         for x in dir(args) if x.startswith('mail')])):
-            parser.error(
-                'the stdio reporter was selected but arguments for the mail '
-                'reporter were provided'
-            )
-
-
 def main():
     """This is the main entry point used by setup.cfg."""
     # pylint: disable=protected-access
     try:
-        global retcode
-
         parser = setup_parser()
         args = parser.parse_args()
-        check_args(parser, args)
 
         setup_logging(args.verbose)
 
         # We are gradually migrating away from messing with cfg and passing
         # it everywhere.
-        var_args = vars(args)
-        if var_args['_name'] in ['merge', 'build']:
-            args.func(var_args)
-        else:
-            cfg = load_config(args)
-            args.func(cfg)
+        cfg = load_config(args)
+        retcode = args.func(cfg)
 
         sys.exit(retcode)
     except KeyboardInterrupt:
         print("\nExited at user request.")
         sys.exit(130)
+
+
+if __name__ == '__main__':
```

Comment:
Is this addition needed since the main function is set as entry point in the installation setup?

Reply:
I can remove this with next respin.